node/caching-compiler.js

```diff
@@ -17,8 +17,17 @@ exports.compile = function (content, filename) {
     // Treat the REPL as if there was no filename.
     filename = null;
   }
-  var info = filename ? getPkgInfo(filename) : fallbackPkgInfo;
-  return info ? readWithCache(info, content) : content;
+  
+  try {
+    var info = filename ? getPkgInfo(filename) : fallbackPkgInfo;
+    return info ? readWithCache(info, content) : content;
+  } catch(e) {
+    if (filename) {
+      e.message += 'While processing file: ' + filename;
+    }
+    throw e;
+  }
+
 };
```

Comment:
You probably know way more about javascript optimization, but I thought since https://github.com/v8/v8/commit/9aac80f try-catch didn't have much of an impact anymore. And even when it did, it only concerned code at the same stack-frame, which is almost nothing here.